proto/opamp.proto

```diff
@@ -25,8 +25,8 @@ option go_package = "github.com/open-telemetry/opamp-go/protobufs";
 message AgentToServer {
     // Globally unique identifier of the running instance of the Agent. SHOULD remain
     // unchanged for the lifetime of the Agent process.
-    // Recommended format: https://github.com/ulid/spec
-    string instance_uid = 1;
+    // MUST be 16 bytes long and SHOULD be generated using the UUID v7 spec.
+    bytes instance_uid = 1;
```

Comment:
this change opens up the instance id to contain a string that has non-utf8 encoded characters. Is that something we need to worry about here? I think the answer is no, because we actually do want a byte array to represent a UUID.

Reply:
Yes, that's intentional. We want arbitrary bytes to be valid, not just UTF8 sequences.

Reply:
yep assumed as much, thanks for confirming.